### PR TITLE
Add production react build on master branch - closes #417

### LINKED ISSUE #417: Hide Redux actions on production
When you open up the browser console on http://unleash.x-team.com you can see the state and actions dispatched. This should not be logged in the console on production. 

NODE_ENV is set to `production` there.

### PR DESCRIPTION
**Pull request by nicksp**
I believe that should fix the development react build on production website.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -18,6 +18,7 @@ before_install:
   - if [ "$TRAVIS_BRANCH" == 'develop' ]; then export PATHS_API_URL='http://paths-staging.unleash.x-team.com/api/v1/paths'; fi
   - if [ "$TRAVIS_BRANCH" == 'develop' ]; then export SLACK_BOT_URL='http://bot.unleash.x-team.com'; fi
   - if [ "$TRAVIS_BRANCH" == 'develop' ]; then export APP_NODE_ENV='staging'; fi
+  - if [ "$TRAVIS_BRANCH" == 'master' ]; then export APP_NODE_ENV='production'; fi
```
Comment (karolsojko) [inline .travis.yml]: This won't help actually. 

The way it works is that by default on Travis the `APP_NODE_ENV` is set with `production` value

zrzut ekranu 2017-06-06 16 15 06


it is only overwritten on develop branch to `staging`

Comment (nicksp) [inline .travis.yml]: Haven't guess that after talking to you. Didn't know that

Comment (nicksp) [inline .travis.yml]: Hmm... that's weird then. Node_ENV is correct but that's the only factor that influences react app build. Are we sure that we have production build on a prod site?

Comment (karolsojko) [inline .travis.yml]: Yes, on AWS EC2 Container the value is set properly.

zrzut ekranu 2017-06-07 11 07 52

Did you check if the debug is visible in console, when the container has `NODE_ENV` set to `production` locally? Or is there something more that has to be changed?

Comment (nicksp) [inline .travis.yml]: not yet, going to check that today

Comment (nicksp) [inline .travis.yml]: hrm weird... i tried setting the `NODE_ENV` var locally for the `start` command. What i discovered is that `process.env.NODE_ENV` is undefined regardless of the env variable. Couldn't understand why, but that might be the root cause.

Comment (karolsojko) [inline .travis.yml]: Maybe try setting it temproarily in `docker-compose` file for testing purposes locally https://docs.docker.com/compose/compose-file/#environment

**Comment by szymonmichalak** [comment]
@nicksp I've noticed it's enough to add this:
```js
new webpack.EnvironmentPlugin(['NODE_ENV'])
```
to `plugins` in Webpack config. That should solve the problem. **But** I've also noticed that deployment process of Unleash app is not optimised at all. HMR plugin works on production, JS is not optimised, etc. But that probably should be solved under a separate ticket...

**Comment by szymonmichalak** [comment]
@karolsojko I think we can close this one. Fixed in #473

**Comment by karolsojko** [comment]
closed as suggested

**Pull request closed without merging**